### DISCUSSION TITLE
What is `TWOFA_DEVICE` supposed to be?

### DISCUSSION THREAD

**Discussion by PanxyCris** [gnzsnz/ib-gateway-docker#179] [Q&A] [answered]
Now IBKR requires 2FA, and also add a new `.env` field in this project: 
<img width="870" alt="截屏2024-10-05 下午11 54 26" src="https://github.com/user-attachments/assets/9d516eef-fcc5-42c7-9ea2-d4f20163e2ca">

After checking IBC documentation, it shows just this description:
<img width="620" alt="截屏2024-10-05 下午11 56 16" src="https://github.com/user-attachments/assets/fa44cf1b-6787-42b0-84c4-3052005779dc">

How do I find this `TWOFA_DEVICE`? I've tried serial number, but it still fails:

![截屏2024-10-05 下午11 57 32](https://github.com/user-attachments/assets/2af3120c-9624-433d-8f25-d7ef9f2bebf8)

**Comment by gnzsnz** [comment, marked as answer]
Please look at https://github.com/gnzsnz/ib-gateway-docker/discussions/155#discussioncomment-10590533
